tests/mdev_configuration_allocation_test.go

```diff
@@ -74,6 +53,33 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 		}
 	}
 
+	checkAllMDEVCreated := func(mdevTypeName string, expectedInstancesCount int) func() (*k8sv1.Pod, error) {
+		return func() (*k8sv1.Pod, error) {
+			By(fmt.Sprintf("Checking the number of created mdev types, should be %d of %s type ", expectedInstancesCount, mdevTypeName))
+			check := fmt.Sprintf(`set -x
+		files_num=$(ls -A /sys/bus/mdev/devices/| wc -l)
+		if [[ $files_num != %d ]] ; then
+		  echo "failed, not enough mdevs of type %[2]s has been created"
+		  exit 1
+		fi
+		for x in $(ls -A /sys/bus/mdev/devices/); do
+		  type_name=$(basename "$(readlink -f /sys/bus/mdev/devices/$x/mdev_type)")
+		  if [[ "$type_name" != "%[2]s" ]]; then
+		     echo "failed, not all mdevs of type %[2]s"
+		     exit 1
+		  fi
+		  exit 0
+		done`, expectedInstancesCount, mdevTypeName)
+			testPod := tests.RenderPod("test-all-mdev-created", []string{"/bin/bash", "-c"}, []string{check})
+			testPod, err = virtClient.CoreV1().Pods(util.NamespaceTestDefault).Create(context.Background(), testPod, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			err := wait.PollImmediate(5*time.Second, 3*time.Minute, waitForPod(ThisPod(testPod)))
+			return latestPod, err
```

Comment:
pedantic nit you can ignore: consider making this more explicit:
```golang
if err := wait.PollImmediate(5*time.Second, 3*time.Minute, waitForPod(ThisPod(testPod))); err != nil {
    return nil, err
}
return latestPod, nil
```

Reply:
I think you want to return the pod always as it will be shown in the output. Do you agree?

Reply:
Not sure I do; what's the point of returning `latestPod` when it did error ? Can you trust the data held there ?

Reply:
You will see the phase of the Pod in our case. This will give you a clue on what to debug/look for.